### PR TITLE
[CI][Tests] Add BWC tests for 2.4.0

### PR DESCRIPTION
**Pull request by manasvinibs**
Signed-off-by: Manasvini B Suryanarayana <manasvis@amazon.com>

### Description
Test data generated with:

yarn test:bwc -o https://artifacts.opensearch.org/releases/bundle/opensearch/2.3.0/opensearch-2.3.0-linux-x64.tar.gz -d https://artifacts.opensearch.org/releases/bundle/opensearch-dashboards/2.3.0/opensearch-dashboards-2.3.0-linux-x64.tar.gz -g true

yarn test:bwc -o https://artifacts.opensearch.org/releases/bundle/opensearch/2.3.0/opensearch-2.3.0-linux-x64.tar.gz -d https://artifacts.opensearch.org/releases/bundle/opensearch-dashboards/2.3.0/opensearch-dashboards-2.3.0-linux-x64.tar.gz -g true -s true
 
### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2230
 
### Check List
- [ ] New functionality includes testing.
  - [X] All tests pass
    - [X] `yarn test:jest`
    - [X] `yarn test:jest_integration`
    - [X] `yarn test:ftr`
- [ ] New functionality has been documented.
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by noCharger** [comment]
Do we mean bwc tests for version 2.4.0? (since it's for backward compatibility)

File: .github/workflows/build_and_test_workflow.yml [hunk 1]
```diff
@@ -186,7 +186,7 @@ jobs:
         working-directory: ./artifacts
     strategy:
       matrix:
-        version: [ osd-2.0.0, osd-2.1.0, osd-2.2.0 ]
+        version: [ osd-2.0.0, osd-2.1.0, osd-2.2.0, osd-2.3.0 ]
```
Comment (noCharger) [inline .github/workflows/build_and_test_workflow.yml:189]: Do we need to include 1.3.2? https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2284

Comment (manasvinibs) [reply .github/workflows/build_and_test_workflow.yml:189]: This change is going into main (v2.4.0) for the future changes to be backward compatible with 2.3.0. I see in #2284 1.3.2 was restored into 2.x branch. Do you mean we need to restore 1.3.2 into main?

Comment (kavilla) [reply .github/workflows/build_and_test_workflow.yml:189]: Good question yall. For `main`, it is on version 3.0.0. Via BWC tests we were actually able to realize it is not compatible to migrate directly from `1.x` to `3.x` which is set by OpenSearch reading which version our indices are from and rejecting the migration. So this helps us keep track and ping doc owners to ensure to note this.

tl;dr: 1.3.2 can't be added to main. Users will have to migrate from 1.x to 2.x to 3.x

**Review by kavilla — APPROVED** [review]

**Comment by manasvinibs** [comment]
> Do we mean bwc tests for version 2.4.0? (since it's for backward compatibility)

Good catch!

**Review by kavilla — APPROVED** [review]

**Review by ananzh — APPROVED** [review]

**Review by noCharger — APPROVED** [review]
LGTM. Thanks Mana.
